app/scripts/store.js

```diff
@@ -45,11 +46,16 @@ var exports = (function() {
       this.state.updating = true;
       Promise.all([
         this.fetchModels(),
-        this.fetchScenarios()
+        this.fetchScenarios(),
+        this.fetchModelDetails()
       ])
       .then((jsons) => {
         this.state.models = jsons[0];
         this.state.scenarios = jsons[1];
+        this.state.models = _.map(this.state.models, (m) => {
+          return (m.id === jsons[2].id) ? jsons[2] : m;
```

Comment:
Perhaps a bit more explicit, what is jsons[2]? 
`this.state.details = jsons[2]; `